crypto/bio/b_addr.c

```diff
@@ -853,7 +853,7 @@ int BIO_lookup_ex(const char *host, const char *service, int lookup_type,
             }
 
             if (endp != service && *endp == '\0'
-                    && portnum > 0 && portnum < 65536) {
+                    && portnum >= 0 && portnum < 65536) {
```

Comment:
why this change?  isn't port 0 something special?

Reply:
From the commit log: "Recognize 0 as a valid return value for strtol in BIO_lookup_ex, making platforms without getaddrinfo behave the same as those with getaddrinfo when BIO_lookup_ex() is called with service="0". This change is necessary for test_ssl_old to succeed on platforms without getaddrinfo."

I don't know what port 0 signifies in this context, but it must be valid, or else test_ssl_old wouldn't be calling BIO_lookup_ex("127.0.0.1","0",...).